### PR TITLE
Additional RIDE functions to retrieve data entries on this account imp…

### PR DESCRIPTION
**Pull request by alexeykiselev**
…lemented and tested

### CODE REVIEW

File: pkg/ride/functions_proto.go [hunk 1]
```diff
@@ -310,6 +310,23 @@ func intFromState(env Environment, args ...rideType) (rideType, error) {
...
 }
 
+func intFromSelfState(env Environment, args ...rideType) (rideType, error) {
+	k, err := extractKey(args)
+	if err != nil {
+		return rideUnit{}, nil
+	}
+	a, ok := env.this().(rideAddress)
+	if !ok {
+		return rideUnit{}, nil
```
Comment (esuwu) [inline pkg/ride/functions_proto.go:320]: let's return a specifc error like "failed to convert argument to ride address" or something like this

File: pkg/ride/functions_proto.go [hunk 2]
```diff
@@ -310,6 +310,23 @@ func intFromState(env Environment, args ...rideType) (rideType, error) {
...
+	}
+	r := proto.NewRecipientFromAddress(proto.Address(a))
+	entry, err := env.state().RetrieveNewestIntegerEntry(r, k)
+	if err != nil {
+		return rideUnit{}, nil
```
Comment (esuwu) [inline pkg/ride/functions_proto.go:325]: return err or errors.Wrap(err, ...)

File: pkg/ride/functions_proto.go [hunk 3]
```diff
@@ -310,6 +310,23 @@ func intFromState(env Environment, args ...rideType) (rideType, error) {
 	return rideInt(entry.Value), nil
 }
 
+func intFromSelfState(env Environment, args ...rideType) (rideType, error) {
+	k, err := extractKey(args)
+	if err != nil {
+		return rideUnit{}, nil
```
Comment (esuwu) [inline pkg/ride/functions_proto.go:316]: return err or errors.Wrap(err, ...

File: pkg/ride/functions_proto.go [hunk 4]
```diff
@@ -322,6 +339,23 @@ func bytesFromState(env Environment, args ...rideType) (rideType, error) {
 	return rideBytes(entry.Value), nil
 }
 
+func bytesFromSelfState(env Environment, args ...rideType) (rideType, error) {
+	k, err := extractKey(args)
+	if err != nil {
+		return rideUnit{}, nil
```
Comment (esuwu) [inline pkg/ride/functions_proto.go:345]: why did you decided to not return error we could get?

**Review by esuwu — CHANGES_REQUESTED** [review]

**Comment by esuwu** [comment]
there are many cases where we probably should return an error

File: pkg/ride/generate/main.go [hunk 5]
```diff
@@ -488,13 +488,21 @@ func functionsV5() map[string]string {
...
+	m["1056"] = "booleanFromSelfState"
+	m["1057"] = "bytesFromSelfState"
+	m["1058"] = "stringFromSelfState"
 	m["1081"] = "calculateLeaseID"
 	m["1092"] = "simplifiedLease"
 	m["1093"] = "fullLease"
 	m["LeaseCancel"] = "leaseCancel"
-	m["1054"] = "isDataStorageUntouched"
-	m["1009"] = "hashScriptAtAddress"
+	m["@extrNative(1055)"] = "intValueFromSelfState"
```
Comment (esuwu) [inline pkg/ride/generate/main.go:502]: why do we need both "intValueFromSelfState" and "intFromSelfState"? Don't they do literally the same?

File: pkg/ride/functions_proto.go (hunk 3 above)
Comment (alexeykiselev) [reply pkg/ride/functions_proto.go:316]: No, it's a special class of functions that swallows all errors.

File: pkg/ride/functions_proto.go (hunk 1 above)
Comment (alexeykiselev) [reply pkg/ride/functions_proto.go:320]: Same reason, it have to return `Unit` on every error.

File: pkg/ride/functions_proto.go (hunk 2 above)
Comment (alexeykiselev) [reply pkg/ride/functions_proto.go:325]: Nope

File: pkg/ride/functions_proto.go (hunk 4 above)
Comment (alexeykiselev) [reply pkg/ride/functions_proto.go:345]: Because this functions work like this in Scala node.

File: pkg/ride/generate/main.go (hunk 5 above)
Comment (alexeykiselev) [reply pkg/ride/generate/main.go:502]: Because the first one will return `throw` on `Unit` value, but the latter won't.

**Review by esuwu — APPROVED** [review]

**Pull request merged**